source/elements/oneDPL/source/parallel_api/parallel_range_api.rst

```diff
@@ -46,9 +47,31 @@ of parallel range algorithms.
 .. code:: cpp
 
    // C++20 analogue of std::projected_value_t; exposition only
-   template<typename I, typename Proj>
+   template <typename I, typename Proj>
    using /*projected-value-type*/ = std::remove_cvref_t<std::invoke_result_t<Proj&, std::iter_value_t<I>&>>;
 
+  // C++20 analogue of nothrow-random-access-range proposed for C++26 in P3179R9; exposition only
+  template <typename R>
+  concept nothrow-random-access-range =
+    std::ranges::random_access_range<R> &&
+    std::is_lvalue_reference_v<std::iter_reference_t<std::ranges::iterator_t<R>>> &&
+    std::same_as<std::remove_cvref_t<std::iter_reference_t<std::ranges::iterator_t<R>>>,
+                 std::iter_value_t<std::ranges::iterator_t<R>>>;
+
+Semantic Requirements
+~~~~~~~~~~~~~~~~~~~~~
+
+A type ``R`` models ``nothrow-random-access-range`` if no exceptions are thrown from:
+
+- increment, decrement, copy construction, move construction, copy assignment, move assignment,
+  comparisons or indirection through valid iterators of type ``std::ranges::iterator_t<R>``;
+- ``-`` operator, copy construction, move construction, copy assignment, move assignment,
+  or comparisons between valid values of type ``std::ranges::iterator_t<R>``;
+- ``-``, ``+``, ``-=``, ``+=``, ``[]`` operators on valid values of type
+  ``std::ranges::iterator_t<R>`` and ``std::iter_difference_t<std::ranges::iterator_t<R>>``;
```

Comment:
```suggestion
- any operation on an object of type ``std::ranges::iterator_t<R>`` required by the ``std::random_access_iterator`` concept;
```
While the standard says that some operations on no-throw iterators are allowed to throw, after close examination the only one I found not covered by the list is the default constructor. I think we can either ignore it or mention as an exception.